### ISSUE TITLE
memleak error on  5.0.10-1.el7.elrepo.x86_64

### ISSUE THREAD

**Issue by lemontree8801** [iovisor/bcc#2329]
my kernel (CentOS) is
`5.0.10-1.el7.elrepo.x86_64`
And installed bcc-tools following,
```
# yum install -y bcc-tools
# rpm -qa | grep bcc
bcc-tools-0.6.1-2.el7.x86_64
python-bcc-0.6.1-2.el7.x86_64
bcc-0.6.1-2.el7.x86_64
```
But memleak fail with following
```
# memleak -a -p 6548
In file included from <built-in>:3:
In file included from /virtual/include/bcc/helpers.h:23:
In file included from /lib/modules/5.0.10-1.el7.elrepo.x86_64/build/include/linux/log2.h:16:
In file included from /lib/modules/5.0.10-1.el7.elrepo.x86_64/build/include/linux/bitops.h:19:
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/bitops.h:209:9: error: 'asm goto' constructs are not supported yet
        return GEN_BINARY_RMWcc(LOCK_PREFIX __ASM_SIZE(bts), *addr, c, "Ir", nr);
               ^
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/rmwcc.h:60:32: note: expanded from macro 'GEN_BINARY_RMWcc'
#define GEN_BINARY_RMWcc(X...) RMWcc_CONCAT(GEN_BINARY_RMWcc_, RMWcc_ARGS(X))(X)
                               ^
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/rmwcc.h:10:28: note: expanded from macro 'RMWcc_CONCAT'
#define RMWcc_CONCAT(a, b) __RMWcc_CONCAT(a, b)
                           ^
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/rmwcc.h:9:30: note: expanded from macro '__RMWcc_CONCAT'
#define __RMWcc_CONCAT(a, b) a ## b
                             ^
note: (skipping 2 expansions in backtrace; use -fmacro-backtrace-limit=0 to see all)
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/rmwcc.h:54:2: note: expanded from macro 'GEN_BINARY_RMWcc_6'
        __GEN_RMWcc(op " %[val], " arg0, var, cc,                       \
        ^
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/rmwcc.h:21:2: note: expanded from macro '__GEN_RMWcc'
        asm_volatile_goto (fullop "; j" #cc " %l[cc_label]"             \
        ^
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/include/linux/compiler_types.h:188:37: note: expanded from macro 'asm_volatile_goto'
#define asm_volatile_goto(x...) asm goto(x)
                                    ^
In file included from <built-in>:3:
In file included from /virtual/include/bcc/helpers.h:23:
In file included from /lib/modules/5.0.10-1.el7.elrepo.x86_64/build/include/linux/log2.h:16:
In file included from /lib/modules/5.0.10-1.el7.elrepo.x86_64/build/include/linux/bitops.h:19:
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/bitops.h:255:9: error: 'asm goto' constructs are not supported yet
        return GEN_BINARY_RMWcc(LOCK_PREFIX __ASM_SIZE(btr), *addr, c, "Ir", nr);
               ^
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/rmwcc.h:60:32: note: expanded from macro 'GEN_BINARY_RMWcc'
#define GEN_BINARY_RMWcc(X...) RMWcc_CONCAT(GEN_BINARY_RMWcc_, RMWcc_ARGS(X))(X)
                               ^
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/rmwcc.h:10:28: note: expanded from macro 'RMWcc_CONCAT'
#define RMWcc_CONCAT(a, b) __RMWcc_CONCAT(a, b)
                           ^
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/rmwcc.h:9:30: note: expanded from macro '__RMWcc_CONCAT'
#define __RMWcc_CONCAT(a, b) a ## b
                             ^
note: (skipping 2 expansions in backtrace; use -fmacro-backtrace-limit=0 to see all)
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/rmwcc.h:54:2: note: expanded from macro 'GEN_BINARY_RMWcc_6'
        __GEN_RMWcc(op " %[val], " arg0, var, cc,                       \
        ^
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/rmwcc.h:21:2: note: expanded from macro '__GEN_RMWcc'
        asm_volatile_goto (fullop "; j" #cc " %l[cc_label]"             \
        ^
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/include/linux/compiler_types.h:188:37: note: expanded from macro 'asm_volatile_goto'
#define asm_volatile_goto(x...) asm goto(x)
                                    ^
In file included from <built-in>:3:
In file included from /virtual/include/bcc/helpers.h:23:
In file included from /lib/modules/5.0.10-1.el7.elrepo.x86_64/build/include/linux/log2.h:16:
In file included from /lib/modules/5.0.10-1.el7.elrepo.x86_64/build/include/linux/bitops.h:19:
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/bitops.h:308:9: error: 'asm goto' constructs are not supported yet
        return GEN_BINARY_RMWcc(LOCK_PREFIX __ASM_SIZE(btc), *addr, c, "Ir", nr);
               ^
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/rmwcc.h:60:32: note: expanded from macro 'GEN_BINARY_RMWcc'
#define GEN_BINARY_RMWcc(X...) RMWcc_CONCAT(GEN_BINARY_RMWcc_, RMWcc_ARGS(X))(X)
                               ^
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/rmwcc.h:10:28: note: expanded from macro 'RMWcc_CONCAT'
#define RMWcc_CONCAT(a, b) __RMWcc_CONCAT(a, b)
                           ^
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/rmwcc.h:9:30: note: expanded from macro '__RMWcc_CONCAT'
#define __RMWcc_CONCAT(a, b) a ## b
                             ^
note: (skipping 2 expansions in backtrace; use -fmacro-backtrace-limit=0 to see all)
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/rmwcc.h:54:2: note: expanded from macro 'GEN_BINARY_RMWcc_6'
        __GEN_RMWcc(op " %[val], " arg0, var, cc,                       \
        ^
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/rmwcc.h:21:2: note: expanded from macro '__GEN_RMWcc'
        asm_volatile_goto (fullop "; j" #cc " %l[cc_label]"             \
        ^
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/include/linux/compiler_types.h:188:37: note: expanded from macro 'asm_volatile_goto'
#define asm_volatile_goto(x...) asm goto(x)
                                    ^
In file included from /virtual/main.c:2:
In file included from /lib/modules/5.0.10-1.el7.elrepo.x86_64/build/include/uapi/linux/ptrace.h:107:
In file included from /lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/ptrace.h:94:
In file included from /lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/paravirt_types.h:46:
In file included from /lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/nospec-branch.h:6:
In file included from /lib/modules/5.0.10-1.el7.elrepo.x86_64/build/include/linux/static_key.h:1:
In file included from /lib/modules/5.0.10-1.el7.elrepo.x86_64/build/include/linux/jump_label.h:117:
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/jump_label.h:23:2: error: 'asm goto' constructs are not supported yet
        asm_volatile_goto("1:"
        ^
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/include/linux/compiler_types.h:188:37: note: expanded from macro 'asm_volatile_goto'
#define asm_volatile_goto(x...) asm goto(x)
                                    ^
In file included from /virtual/main.c:2:
In file included from /lib/modules/5.0.10-1.el7.elrepo.x86_64/build/include/uapi/linux/ptrace.h:107:
In file included from /lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/ptrace.h:94:
In file included from /lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/paravirt_types.h:46:
In file included from /lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/nospec-branch.h:6:
In file included from /lib/modules/5.0.10-1.el7.elrepo.x86_64/build/include/linux/static_key.h:1:
In file included from /lib/modules/5.0.10-1.el7.elrepo.x86_64/build/include/linux/jump_label.h:117:
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/arch/x86/include/asm/jump_label.h:39:2: error: 'asm goto' constructs are not supported yet
        asm_volatile_goto("1:"
        ^
/lib/modules/5.0.10-1.el7.elrepo.x86_64/build/include/linux/compiler_types.h:188:37: note: expanded from macro 'asm_volatile_goto'
#define asm_volatile_goto(x...) asm goto(x)
                                    ^
5 errors generated.
Traceback (most recent call last):
  File "/usr/share/bcc/tools/memleak", line 401, in <module>
    bpf = BPF(text=bpf_source)
  File "/usr/lib/python2.7/site-packages/bcc/__init__.py", line 318, in __init__
    raise Exception("Failed to compile BPF text")
Exception: Failed to compile BPF text
```
What am I doing wrong?

**Comment by yonghong-song** [comment]
This is a known issue with 5.0 kernel. The following commit fixed the issue
```
commit a753e5725234c041b772577cddd21b7a39b815a1
Author: yonghong-song <ys114321@gmail.com>
Date:   Thu Jan 10 21:10:56 2019 -0800

    workaround kernel 5.0 asm_volatile_goto issue (#2133)
    
    Fix issue #2119.
    
    Kernel 5.0 added more usages of asm goto and llvm
    does not support asm goto yet. This resulted in
    compilation error for virtually any bcc scripts.
    
    The workaround here is to redefine asm_volatile_goto to
      asm volatile("invalid use of asm_volatile_goto")
    which can pass clang.
    If bpf program does not use asm_volatile_goto,
    nothing bad will happen. The functions using
    asm_volatile_goto will be thrown away.
    If bpf program accidentally uses asm_volatile_goto,
    a compilation error like below will be printed out:
      <inline asm>:1:2: error: invalid register/token name
              invalid use of asm_volatile_goto
              ^
      LLVM ERROR: Error parsing inline asm
    
    Signed-off-by: Yonghong Song <yhs@fb.com>
```

Please try bcc 0.8.0 or 0.9.0 or latest trunk.

**Comment by lemontree8801** [comment]
thanks~
